code401Challenges/src/main/java/code401Challenges/hashtable/Hashtable.java

```diff
@@ -0,0 +1,60 @@
+package code401Challenges.hashtable;
+
+
+public class Hashtable {
+
+    private Node[] map;
+
+
+    //constructor
+    public Hashtable(int size){
+        map = new Node[size];
+    }
+
+    //methods
+    public int hash(String key){
+        int hashValue = 0;
+        char[] words = key.toCharArray();
+        for(int i = 0; i < words.length; i++){
+            hashValue += words[i];
+        }
+        hashValue = (hashValue * 1069) % map.length;
+        return hashValue;
+    }
+
+    public void add(String key, String value){
+        Node node = new Node(key,value);
+        int hashKey = hash(key);
+        if(this.map[hashKey] == null){
+            map[hashKey] = node;
+        }
+        else {
+            Node temp = map[hashKey];
+            map[hashKey] = node;
+            map[hashKey].setNext(temp);
+        }
+    }
+
+    public String get(String key){
+        int searchHashKey = hash(key);
+        while(this.map[searchHashKey] != null){
+            if(this.map[searchHashKey].getKey() == key){
```

Comment:
You should definitely use `.equals()` instead of `==`. Remember, `==` checks to see if the two things being compared a literally the same (the same thing stored in the same place in memory). `.equals()` checks to see if things are logically equal -the things being compared have the same value but aren't literally the same thing stored in memory.

This probably still passed your tests because Java has a built in behavior when using hard coded Strings where it uses the exact same string, stored in the same place in memory instead of creating 2 string and storing them separately.